contracts/protocol/clients/DRFeeMutualizer.sol

```diff
@@ -0,0 +1,559 @@
+     */
+    function newAgreement(
+        uint256 _sellerId,
+        address _tokenAddress,
+        uint256 _disputeResolverId,
+        uint256 _maxAmountPerTx,
+        uint256 _maxAmountTotal,
+        uint256 _timePeriod,
+        uint256 _premium,
+        bool _refundOnCancel
+    ) external onlyOwner returns (uint256 agreementId) {
+        if (_sellerId == 0) revert InvalidSellerId();
+        if (_maxAmountPerTx == 0) revert MaxAmountPerTxMustBeGreaterThanZero();
+        if (_maxAmountTotal < _maxAmountPerTx) revert MaxTotalMustBeGreaterThanOrEqualToMaxPerTx();
+        if (_timePeriod == 0) revert TimePeriodMustBeGreaterThanZero();
+
+        uint256 existingAgreementId = sellerToTokenToDisputeResolverToAgreement[_sellerId][_tokenAddress][
+            _disputeResolverId
+        ];
+        if (existingAgreementId != 0) {
+            Agreement storage existingAgreement = agreements[existingAgreementId];
+            if (
+                existingAgreement.startTime > 0 &&
+                !existingAgreement.isVoided &&
+                block.timestamp <= existingAgreement.startTime + existingAgreement.timePeriod
+            ) {
+                revert AgreementAlreadyExists();
+            }
+        }
+
+        agreementId = agreements.length;
+
+        agreements.push(
+            Agreement({
+                maxAmountPerTx: _maxAmountPerTx,
+                maxAmountTotal: _maxAmountTotal,
+                timePeriod: _timePeriod,
+                premium: _premium,
+                tokenAddress: _tokenAddress,
+                refundOnCancel: _refundOnCancel,
+                isVoided: false,
+                startTime: 0,
+                totalMutualized: 0,
+                sellerId: _sellerId
+            })
+        );
+
+        sellerToTokenToDisputeResolverToAgreement[_sellerId][_tokenAddress][_disputeResolverId] = agreementId;
+
+        emit AgreementCreated(agreementId, _sellerId, _tokenAddress, _disputeResolverId);
+    }
+
+    /**
+     * @notice Voids an existing agreement
+     * @param _agreementId The ID of the agreement to void
+     * @dev Can be called by the seller or owner (if refundOnCancel is true). Calculates time-based refunds.
+     *
+     * Reverts if:
+     * - agreementId is invalid
+     * - Agreement is already voided
+     * - Caller is not authorized
+     * - ERC20 or native currency transfer fails
+     * - Seller not found
+     */
+    function voidAgreement(uint256 _agreementId) external {
+        if (_agreementId == 0 || _agreementId >= agreements.length) revert InvalidAgreementId();
+        bool premiumRefunded;
+        Agreement storage agreement = agreements[_agreementId];
+        if (agreement.isVoided) revert AgreementAlreadyVoided();
+
+        (bool exists, BosonTypes.Seller memory seller, ) = IBosonAccountHandler(BOSON_PROTOCOL).getSeller(
+            agreement.sellerId
+        );
+        if (!exists) revert SellerNotFound();
```

Comment:
This means it's not possible to void an agreement created for an non existing seller.
However, it is currently allowed to create an agreement for a non existing seller.
Not a major problem IMHO, but seems a bit inconsistent